Library/Source/NapiBridge.h

```diff
@@ -19,22 +19,22 @@ namespace babylon
             Definition& operator=(const Definition&) = delete;
 
             template<void(ImplT::*method)(const Napi::CallbackInfo& info)>
-            Definition& AddVoidReturningMethod(const std::string& name)
+            Definition& AddVoidReturningMethod(const char* name)
```

Comment:
I did that because a string is created/allocation is made and only used thru its c_str in AddVoidReturningMethod and AddValueReturningMethod.

Reply:
It was being passed by ref, so I'm pretty sure it shouldn't have been copying. The code as it was should have been technically equivalent to "const std::string*", just safer because it uses refs. Unless I'm misunderstanding something?

Reply:
yep, a string is allocated from a const char* then passed as reference and its const c pointer is used. string is deallocated after the code block.

Reply:
see the difference here https://www.godbolt.org/z/F6vfmv

Reply:
Oh, you mean allocated on the _calling_ side, since this method is called overwhelmingly using string literals. Gotcha. Yes, that will allocate an std::string on the calling side, then discard it once it's done. That's not something I'm super worried about since it's cheap, one time, and preserves the convention of avoiding raw pointers when possible; but in this case, I can also see the argument for raw pointers since this method will be pretty much only called using literals. Totally up to you on this one.

Reply:
Actually, since this is overwhelmingly done using string literals... Any chance we could constexpr this? Might be a long shot considering what's happening under the hood, but I can't think of a reason there's be compile-time ambiguity about what's going on here. Probably not worth it for this code review -- and maybe not at all, I can't think of any real runtime benefit. Might be fun, though. :smiley: